core/src/subgraph/instance_manager.rs

```diff
@@ -665,6 +676,23 @@ where
 
             match res {
                 Ok(needs_restart) => {
+                    if should_try_update_sync_status(
+                        synced,
+                        sync_status_timer,
+                        *SYNC_STATUS_THRESHOLD,
+                        // This is the new deployment head since the block just has been processed.
+                        &block_ptr,
+                        || chain_store.chain_head_ptr(),
```

Comment:
Passing the chain store unfortunately makes the test:

- depend on the database/store
- run slower

I ended up going for easier/faster testability because, as mentioned before, I really wanted to be sure the boolean part is correct, so we don't have to do patches on this.

If you think it really damages the readability, I can change this.

Reply:
I agree testability is more important here. You could mock the `ChainStore` trait but that would be too much boilerplate unless you split this method to a separate trait. So ok, we can keep this as is.